invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -24,24 +24,30 @@
 class SearchFieldTransformer(TreeTransformer):
     """Transform from user-friendly field names to internal field names."""
 
-    def __init__(self, mapping, *args, **kwargs):
+    def __init__(self, mapping, safelist, *args, **kwargs):
         """Constructor."""
         self._mapping = mapping
+        self._safelist = safelist
         super().__init__(self, *args, **kwargs)
 
     @classmethod
-    def factory(cls, mapping=None):
+    def factory(cls, mapping=None, safelist=None):
         """Create a new field transformer."""
-        return partial(cls, mapping or {})
+        return partial(cls, mapping or {}, safelist or [])
 
     def visit_search_field(self, node, context):
         """Visit a search field."""
-        if node.name not in self._mapping:
+        # Use the node name if not mapped for transformation.
+        term_name = self._mapping.get(node.name, node.name)
+
+        # If a safelist exists, the term must be safelisted to be queried.
+        if self._safelist and not term_name in self._safelist:
```

Comment:
Do we have a case where we would not want a safelist? records? This is to make our lives easier when defining the parsing?

Reply:
Sorry for the late reply!

Yes there are searches that we do not want / need a safelist. Initially the terms mapping also acted as a safelist and that was particularly useful for users becasuse some terms are not meant to be queried.

However, since we shipped two features in one (mapping + safelisting) it also meant that if a transform map was defined, then any other terms are considered were blocked (e.g. can't be queried at all). In fact, this had a side effect on which the transformation map needed things like `{"email": "email"}` since the term was not transformed AND being blocked if not defined.

TL;DR : yes we do have cases where we do not want to have a safelist. And yes, this separation of features makes our lives easier.